src/main/java/com/iexec/sms/secret/Secret.java

```diff
@@ -46,16 +43,24 @@ public class Secret {
     private boolean isEncryptedValue;
 
     /* Clear secrets at construction */
-    public Secret(String address, String value) {
+    public Secret(String address, String value, boolean isEncryptedValue) {
         this.address = address;
-        this.setValue(value, false);
-    }
-
-    public void setValue(String value, boolean isEncryptedValue) {
         this.value = value;
         this.isEncryptedValue = isEncryptedValue;
     }
 
+    /**
+     * Copies the current {@link Secret} object,
+     * while replacing the old value with the new one.
+     *
+     * @param newValue         Value to use for new object.
+     * @param isEncryptedValue Whether this value is encrypted.
+     * @return A new {@link Secret} object with new value.
+     */
+    public Secret withValue(String newValue, boolean isEncryptedValue) {
```

Comment:
Maybe 2 methods to avoid confusions:
```
withEncryptedValue
withDecryptedValue
```

Reply:
See https://github.com/iExecBlockchainComputing/iexec-sms/pull/134/commits/16156ec6a8394e6ad04b0c2a4332e63004f837b7.